astropy/table/tests/test_masked.py

```diff
@@ -414,3 +414,19 @@ def test_coercing_fill_value_type():
     c.set_fill_value('0')
     c2 = MaskedColumn(c, dtype=np.int32)
     assert isinstance(c2.fill_value, np.int32)
+
+
+def test_mask_copy():
+    """Test that the mask is copied when copying a table (issue #7362)."""
+
+    t = Table([[2, 4, 1, 3], [3., 2., np.nan, 4]],
```

Comment:
This test is correct (and reflects the original issue) but somehow is more obfuscated than need be.
```
c = MaskedColumn([1, 2], mask=[False, True])
c2 = MaskedColumn(c)
c2.mask[0] = True
assert np.all(c.mask == [False, True])
```

Reply:
Indeed, I simplified with your suggestions, thanks.